commit e783771947cbe99c63661609e92147c7152f851d
Author: Eric Anholt <eric@anholt.net>
Date:   Mon Feb 5 18:53:18 2018 +0000

    drm/vc4: Don't wait for vblank on fkms cursor updates.
    
    We don't use the same async update path between fkms and normal kms,
    and the normal kms workaround ended up making us wait.  This became a
    larger problem in rpi-4.14.y, as the USB HID update rate throttling
    got (accidentally?) dropped.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/gpu/drm/vc4/vc4_kms.c b/drivers/gpu/drm/vc4/vc4_kms.c
index 50c4959b5bd3..f73343e08fe6 100644
--- a/drivers/gpu/drm/vc4/vc4_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_kms.c
@@ -53,7 +53,8 @@ vc4_atomic_complete_commit(struct drm_atomic_state *state)
 	 * display lists before we free it and potentially reallocate
 	 * and overwrite the dlist memory with a new modeset.
 	 */
-	state->legacy_cursor_update = false;
+	if (!vc4->firmware_kms)
+		state->legacy_cursor_update = false;
 
 	drm_atomic_helper_commit_hw_done(state);
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1190_e783771947cbe99c63661609e92147c7152f851d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1190_e783771947cbe99c63661609e92147c7152f851d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
